pkg/config/validation/validation.go

```diff
@@ -84,6 +84,7 @@ var (
 		"5xx",
 		"gateway-error",
 		"reset",
+		"reset-before-request",
```

Comment:
should check the proxy version before applying to it

Reply:
This is for the VirtualService validation right? IIUC, that's not proxy dependent. If there's even 1 proxy with the latest version, it's valid. Or is your point that if there's a single proxy on an old version, that proxy's RDS will get NACK'd? L

Reply:
If set this after Istio upgrade and no proxy restart (proxy running old version) - wouldn't this cause RDS NACK?

Reply:
just put a hold to resolve this comment

Reply:
If a user sets it yes; all this does is not block a VS setting this new value. Adding a version gate here stops user error but how would we even write a validation to check all of the proxy versions and conditionally allow the VS?

Reply:
> Adding a version gate here stops user error but how would we even write a validation to check all of the proxy versions and conditionally allow the VS?

Yeah. I wasn't suggesting to add validation check. Just want to understand how it works. Just looked at the Envoy [code](https://github.com/envoyproxy/envoy/blob/main/source/common/router/retry_state_impl.cc#L190)  - It does not NACK. It uses the validation only in the strict header check mode when retry on is passed via x-envoy-retry-on header. So we are good here

Reply:
If it did NACK, the fix IMO should be to silently remove it for old proxies if anything. Good to know nothing is needed though.

validation.go cannot be proxy dependant though

Reply:
But this would fail older proxy connecting with new isitod

Reply:
Why would it fail?